api/src/classes/company.py

```diff
@@ -0,0 +1,154 @@
+        """Return Companies from the database and return them, with pagination.
+
+        :param session: Database session.
+        :param params: Pagination parameters.
+        :return: Paginated response to return.
+        """
+        q: Any = select(Company)
+        q = q.order_by(desc(Company.networth)) if not params.ascending else q.order_by(Company.networth)
+
+        paginator: Paginate = Paginate(query=q, session=session, params=params)
+
+        res: list[dict[str, Any]] = [
+            {
+                "name": company.name,
+                "owner_id": company.owner_id,
+                "networth": company.networth,
+                "is_bankrupt": company.is_bankrupt,
+            }
+            for company in paginator.get_data()
+        ]
+
+        # If nothing
+        if not res:
+            raise HTTPException(status_code=404, detail="Companies not found.")
+
+        paginator.add_data({"companies": res})
+        return paginator.get_page()
+
+    @classmethod
+    def create_company(cls, session: Session, data: CompanyCreate) -> "CompanyRepresentation":
+        """Return instance with newly created Company.
+
+        If the `owner` already has a company, they cannot create a new one.
+        If the `name` has been taken, the company cannot be created.
+
+        :param session: Database session.
+        :param data: Company create data.
+        :return: Instance with the created company.
+        """
+        # Query the database
+        target: Sequence[Company] = session.exec(
+            select(Company).where(or_(Company.name == data.name, Company.owner_id == data.owner_id))
+        ).all()
+
+        if target and not any(x.is_bankrupt for x in target):
+            raise HTTPException(status_code=409, detail="Such company already exists.")
+
+        try:
+            new_company: Company = Company(name=data.name, owner_id=data.owner_id)
+            session.add(new_company)
+            session.flush()
+            session.commit()
+            session.refresh(new_company)
+
+        except SQLAlchemyError:
+            session.rollback()
+            raise HTTPException(status_code=500, detail="Unable to create a new Company") from None
+
+        return cls(session=session, company=new_company)
+
+    def get_company(self) -> Company | None:
+        """Get the Company model bound to the instance.
+
+        :return: Bound Company model.
+        """
+        return self.company
+
+    def get_details(self) -> CompanyPublic | None:
+        """Get the details of the Company.
+
+        :return: Company details.
+        """
+        if self.company is not None:
+            return CompanyPublic.model_validate(self.company)
+        return None
+
+    def update(self, data: CompanyUpdate) -> None:
+        """Update the Company's details.
+
+        :param data: Data of Company to update.
+        :return: None
+        """
+        try:
+            has_changed: bool = False
```

Comment:
Is the `: bool` needed?

Reply:
No, not needed. Will provide a reason as to why I did that later on

Reply:
I have come from a strict background, so I am used to adding things like `: bool` for trivial variables, like this one. It helps get a good understanding of that variable's purpose. Also done to improve readability.

I can remove it, if need be.

Reply:
I feel like it can be inferred that is can only be a bool if you're setting it equal to False.
Personally, I would only add a type hint there if it could be something other than a bool e.g. `foo: bool | None = False`